libs/fslock/lock_unix.go

```diff
@@ -0,0 +1,48 @@
+//go:build darwin || freebsd || linux
+
+package fslock
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"syscall"
+)
+
+func (l *Locker) lock() (err error) {
+	l.file, err = os.OpenFile(l.path, os.O_CREATE|os.O_RDWR, 0666)
+	if err != nil {
+		return fmt.Errorf("fslock: error opening file: %v", err)
+	}
+
+	_, err = l.file.WriteString(strconv.Itoa(os.Getpid()))
+	if err != nil {
+		return fmt.Errorf("fslock: error writing process id: %v", err)
+	}
+
+	err = syscall.Flock(int(l.file.Fd()), syscall.LOCK_EX|syscall.LOCK_NB)
+	if err != nil && err.Error() == "resource temporarily unavailable" {
```

Comment:
This is not how you compare errors properly. Isn't there are a way to use `errors.Is` instead?

Reply:
@liamsi, in such a case I would need a concrete error defined somewhere to compare it to. Unfortunately, there is no such an error for this case, as this is an API on the boundaries between Go's runtime and OS, which does not understand Go's semantics.

Resolved?

Reply:
Yes, maybe add a comment so the next person has an explanation though.